src/environment/shell.go

```diff
@@ -800,37 +810,84 @@ func isPathSeparator(env Environment, c uint8) bool {
 // Trailing path separators are removed before extracting the last element.
 // If the path consists entirely of separators, Base returns a single separator.
 func Base(env Environment, path string) string {
-	if path == "/" {
-		return path
-	}
 	volumeName := filepath.VolumeName(path)
 	// Strip trailing slashes.
-	for len(path) > 0 && isPathSeparator(env, path[len(path)-1]) {
+	for len(path) > 0 && IsPathSeparator(env, path[len(path)-1]) {
 		path = path[0 : len(path)-1]
 	}
+	if len(path) == 0 {
+		return env.PathSeparator()
+	}
 	if volumeName == path {
 		return path
 	}
 	// Throw away volume name
 	path = path[len(filepath.VolumeName(path)):]
 	// Find the last element
 	i := len(path) - 1
-	for i >= 0 && !isPathSeparator(env, path[i]) {
+	for i >= 0 && !IsPathSeparator(env, path[i]) {
 		i--
 	}
 	if i >= 0 {
 		path = path[i+1:]
 	}
 	// If empty now, it had only slashes.
-	if path == "" {
+	if len(path) == 0 {
 		return env.PathSeparator()
 	}
 	return path
 }
 
+// ParsePath parses an input path and returns a clean root and a clean path.
+func ParsePath(env Environment, inputPath string) (root, path string) {
+	if len(inputPath) == 0 {
+		return "", ""
```

Comment:
I think you can just return the variables as well, no?
```suggestion
		return root, path
```

Reply:
Yes, of course. And even simpler, a naked `return` is all we need here.